doc/object_properties.adoc

```diff
@@ -0,0 +1,145 @@
+= Object Properties
+
+WARNING: There may be missing/incorrect information here, please refer to the https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L7141-L7304[Lua API] and the https://github.com/minetest/minetest/blob/master/src/script/common/c_content.cpp#L186-L442[relevant engine code]
+
+WARNING: This is missing the default values for everything
+
+== `player` only
+
+=== hp_max
+
+Sets the max HP of a player. Wanted value is `u16`.
```

Comment:
What if u16 is exceeded? Does it overflow (wrap around) or is it clamped?

Reply:
im not sure what the code actually does about this as I'm not very experienced in C++, you can see the relevant line: https://github.com/minetest/minetest/blob/master/src/script/common/c_content.cpp#L198

Reply:
```cpp
#define rangelim(d, min, max) ((d) < (min) ? (min) : ((d) > (max) ? (max) : (d)))
```
Looks like it's clamped

Reply:
should i add a little bit to the end of the sentence mentioning that?